corehq/apps/accounting/utils/invoicing.py

```diff
@@ -0,0 +1,77 @@
+import datetime
+
+from django.db.models import Q, Sum
+
+from corehq.apps.accounting.models import (
+    Invoice,
+    SubscriptionType,
+    CustomerInvoice,
+)
+
+UNPAID_INVOICE_THRESHOLD = 100
+
+
+def _get_all_unpaid_saas_invoices():
+    return Invoice.objects.filter(
+        is_hidden=False,
+        subscription__service_type=SubscriptionType.PRODUCT,
+        date_paid__isnull=True,
+    )
+
+
+def _get_unpaid_saas_invoices_in_downgrade_daterange(today):
+    return _get_all_unpaid_saas_invoices().filter(
+        date_due__lte=today - datetime.timedelta(days=1)
+    ).order_by('date_due').select_related('subscription__subscriber')
+
+
+def get_unpaid_invoices_over_threshold_by_domain(today, domain):
```

Comment:
I realize your just moving this, and also that this is a base PR into which you're merging individually reviewed PRs, but in reviewing one of the PRs I got confused about what this function does and what it returns. From the name I though it returned unpaid invoices over some threshold, but it actually returns the oldest unpaid invoice whose amount is over some threshold, as well as the amount due in that invoice, as a tuple.

All that to say, might be worth changing the name of this function to be
`get_oldest_unpaid_invoice_over_threshold` or something like that

Reply:
@dannyroberts https://github.com/dimagi/commcare-hq/pull/27196